src/main/java/io/rcktapp/api/handler/firehose/DeliveryStreamNameSpliterator.java

```diff
@@ -0,0 +1,53 @@
+package io.rcktapp.api.handler.firehose;
+
+import com.amazonaws.services.kinesisfirehose.AmazonKinesisFirehose;
+import com.amazonaws.services.kinesisfirehose.model.ListDeliveryStreamsRequest;
+
+import java.util.List;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class DeliveryStreamNameSpliterator implements Spliterator<String> {
+    private final AmazonKinesisFirehose firehoseClient;
+    private String last = null;
+
+    private DeliveryStreamNameSpliterator(AmazonKinesisFirehose firehoseClient) {
+        this.firehoseClient = firehoseClient;
+    }
+
+    static Stream<String> stream(AmazonKinesisFirehose firehoseClient) {
+        return StreamSupport.stream(new DeliveryStreamNameSpliterator(firehoseClient), false);
+    }
```

Comment:
This is the entry point to grab a stream - a static you hand the firehose client to.